### DISCUSSION TITLE
Possible  bug in latest develop branch with replaying macros and finding ‘z;’ and infinite loops...

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#1732] [General]
I don’t have a reproducible case I can share yet, and it might be at least a  week out before I do. Reporting in case someone else is seeing similar problems with develop .

I have this macro that one of the steps is the addcol-shell command `z;` that the macro says there is no such command.  I try running the macro a second time and Visidata gets into a busy infinite loop, but it is not telling me anything.  It is busy doing things, just not responding to input, and not printing anything. 

 I can run the ‘z;’ before and after the running the macro the first time so I know the key binding works.  Not sure if this is because it is running from macro playback.  It is able to run other commands in the macro, not sure why 'z;' is at issue.

It looks like the failure is being reported here: https://github.com/saulpw/visidata/blob/develop/visidata/mainloop.py#L309

**Comment by saulpw** [comment]
Okay, keep us posted.  If you can get a stack trace or anything we can look into it.
